internal/provider/resource/dynamic_secret/dynamic_secret_kubernetes.go

```diff
@@ -372,23 +376,29 @@ func NewDynamicSecretKubernetesResource() resource.Resource {
 				return types.ObjectNull(map[string]attr.Type{}), diags
 			}
 
-			var audiences []string
-			for i, v := range audiencesRaw {
-				s, ok := v.(string)
-				if !ok {
-					diags.AddError(
-						"Invalid audience element type",
-						"Expected audience at index "+string(rune(i))+" to be a string but got something else.",
-					)
-					return types.ObjectNull(map[string]attr.Type{}), diags
+			plannedListIsUndefinedOrEmpty := currentState.Audiences.IsNull() || len(currentState.Audiences.Elements()) == 0
+
+			if len(audiencesRaw) == 0 && plannedListIsUndefinedOrEmpty {
+				configuration["audiences"] = currentState.Audiences
+			} else {
+				var audiences []string
+				for i, v := range audiencesRaw {
+					s, ok := v.(string)
+					if !ok {
+						diags.AddError(
+							"Invalid audience element type",
+							"Expected audience at index "+string(rune(i))+" to be a string but got something else.",
+						)
+						return types.ObjectNull(map[string]attr.Type{}), diags
+					}
+					audiences = append(audiences, s)
 				}
-				audiences = append(audiences, s)
-			}
 
-			audiencesList, listDiags := types.ListValueFrom(ctx, types.StringType, audiences)
-			diags.Append(listDiags...)
+				audiencesList, listDiags := types.ListValueFrom(ctx, types.StringType, audiences)
+				diags.Append(listDiags...)
```

Comment:
I guess we're also missing a check for diagnostic errors here.. Can you please do it for the entire file?